app/server/api/export_api.py

```diff
@@ -19,365 +19,349 @@
-            _ = ws.cell(column=index + 1, row=1, value="{0}".format(column['header']))
-
-        user_accounts = []
-        credit_transfer_list = []
-
-        # filter user accounts
-        if user_type == 'beneficiary':
-            user_filter = User.has_beneficiary_role
-
-        if user_type == 'vendor':
-            user_filter = User.has_vendor_role
-
-        if date_range == 'all':
-            end_date = datetime.utcnow()
-            start_date = end_date - timedelta(weeks=520)
-
-        if date_range == 'day':
-            # return previous day of transactions
-            end_date = datetime.utcnow()
-            start_date = end_date - timedelta(days=1)
-
-        if date_range == 'week':
-            # return previous week of transactions
-            end_date = datetime.utcnow()
-            start_date = end_date - timedelta(weeks=1)
-
-        if user_filter is not None:
-            user_accounts = User.query.filter(
-                user_filter==True
-            ).options(
-                joinedload(User.transfer_accounts)
-            ).all()
-
-        elif user_type == 'selected':
-            transfer_accounts = TransferAccount.query.filter(TransferAccount.id.in_(selected)).all()
-            user_accounts = [ta.primary_user for ta in transfer_accounts]
-
-        else:
-            user_accounts = User.query.filter(
-                or_(User.has_beneficiary_role == True, User.has_vendor_role == True)
-            ).options(
-                joinedload(User.transfer_accounts)
-            ).all()
-
-        if export_type == 'pdf':
-            file_url = generate_pdf_export(user_accounts, pdf_filename)
-            response_object = {
-                'message': 'Export file created.',
-                'data': {
-                    'file_url': file_url,
-                }
+@standard_executor_job
+def generate_export(post_data):
+    from server.models.credit_transfer import CreditTransfer
+    from server.models.transfer_account import TransferAccount
+    from server.models.user import User
+
+    export_type = post_data.get('export_type')
+    include_transfers = post_data.get('include_transfers')  # True or False
+    include_custom_attributes = post_data.get('include_custom_attributes')  # True or False
+    user_type = post_data.get('user_type')  # Beneficiaries, Vendors, All
+    selected = post_data.get('selected')
+    #  TODO: implement date_range
+    date_range = post_data.get('date_range')  # last day, previous week, or all
+
+    payable_period_type   = post_data.get('payable_period_type', 'month')        # day, week, month
+    payable_period_length = post_data.get('payable_period_length', 1)      # Integer, ie "payable every _2_ months"
+    payable_epoch         = post_data.get('payable_epoch')              # When did the very first cycle begin?
+
+    payable_period_start_date = post_data.get('payable_period_start_date') # any sort of reasonable date string for custom date period
```

Comment:
lol we should really clean this code up at some point... need to spend like two weeks with the only rule being you're only allowed to delete or refactor code.